core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -367,7 +370,11 @@ public RetryOnPrimaryException(StreamInput in) throws IOException {
 
     public interface PrimaryResult<R extends ReplicationRequest<R>> {
 
-        R replicaRequest();
+        /**
+         * Returns the replication request when the primary operation
+         * is successful otherwise returns <code>null</code>
```

Comment:
the description is wrong - we don't know if things were successful or not (and in fact with seq # we will replicate failures). I think we should just say "returns null if no operation needs to be sent to a replica (for example when the operation failed on the primary due to a parsing exception)"